community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -108,9 +110,20 @@
     /**
      * Closes this transaction, committing its changes iff {@link #success()} has been called and
      * {@link #failure()} has NOT been called. Otherwise its changes will be rolled back.
+     *
+     * @return id of the committed transaction or {@link #NOT_COMMITTED} if transaction was rolled back or read-only.
+     */
+    long closeTransaction() throws TransactionFailureException;
+
+    /**
+     * Closes this transaction, committing its changes iff {@link #success()} has been called and
+     * {@link #failure()} has NOT been called. Otherwise its changes will be rolled back.
      */
     @Override
-    void close() throws TransactionFailureException;
+    default void close() throws TransactionFailureException
+    {
+        closeTransaction();
+    }
```

Comment:
Yeah, this is a tough for me as well. At the same time it is kind of nice for KTI to be `AutoCloseable`. I guess we can keep it for now and remove `void close()` later if it causes confusion.